src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -461,14 +465,57 @@ fn all_constructors<'a, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
                 .map(|v| Variant(v.did))
                 .collect()
         }
+        ty::TyChar if exhaustive_integer_patterns => {
+            let (min, max) = (0u128, char::MAX as u128);
+            value_constructors = true;
+            vec![ConstantRange(ty::Const::from_bits(cx.tcx, min, cx.tcx.types.char),
+                               ty::Const::from_bits(cx.tcx, max, cx.tcx.types.char),
+                               RangeEnd::Included)]
+        }
+        ty::TyInt(int_ty) if exhaustive_integer_patterns => {
+            use syntax::ast::IntTy::*;
+            macro_rules! min_max_ty {
+                ($ity:ident, $uty:ty, $sty:expr) => {
+                    ($ity::MIN as $uty as u128, $ity::MAX as $uty as u128, $sty)
+                }
+            }
+            let (min, max, ty) = match int_ty {
+                Isize => min_max_ty!(isize, usize, cx.tcx.types.isize),
```

Comment:
Oh, this is broken, because it's using the host's `usize` / `isize` bitwidth instead of the target's.